tests/test_cases.rs

```diff
@@ -6,18 +6,26 @@ use anyhow::Result;
 use fs_extra::dir::{copy, CopyOptions};
 use std::fs;
 use std::io::{self, Write};
+use log::warn;
 use std::path::Path;
-use std::process::Command;
+use std::process::{Command, exit};
 use tempfile::TempDir;
 
 #[ctor::ctor]
 fn unshorn() {
-    use nix::{mount, sched, unistd};
+    use nix::{errno, mount, sched, unistd};
     use std::os::unix::fs::symlink;
 
     let (uid, gid) = (unistd::geteuid(), unistd::getegid());
     if !uid.is_root() {
-        sched::unshare(sched::CloneFlags::CLONE_NEWUSER).expect("unshare(NEWUSER)");
+        match sched::unshare(sched::CloneFlags::CLONE_NEWUSER) {
+            Ok(_) => {}
+            Err(errno::Errno::EPERM) => {
+                warn!("No permission to unshare namespace, skipping tests");
```

Comment:
`warn!()` doesn't work here (there is no log sink installed) — just use `eprintln!()`.
Also, the message is odd (which namespace?) – how about "unshare(NEWUSER) forbidden and not running as root: skipping tests."?